controller/src/main/java/org/jboss/as/controller/remote/TransactionalProtocolOperationHandler.java

```diff
@@ -155,15 +160,27 @@ public Void run() {
 
                 @Override
                 public void execute(final ManagementRequestContext<ExecuteRequestContext> context) throws Exception {
-                    AccessController.doPrivileged(new PrivilegedAction<Void>() {
-
-                        @Override
-                        public Void run() {
-                            AccessAuditContext.doAs(securityIdentity != null , securityIdentity, remoteAddress, action);
-                            return null;
-                        }
-                    });
-
+                    if (executableRequest.inVmCall && securityIdentity == null) {
```

Comment:
Could that logic be wrapped up inside the Supplier anonymous class's `get` method?

Reply:
Yes, done